pr2_head/src/head.cpp

```diff
@@ -0,0 +1,127 @@
+/**
+ * @file      head.cpp
+ * @brief     Provides basic looking at object frame functionality
+ * @author    Alejandro Bordallo <alex.bordallo@ed.ac.uk>
+ * @date      2016-02-06
+ * @copyright (MIT) 2015 RAD-UoE Informatics
+ */
+
+#include "pr2_head/head.hpp"
+
+Head::Head(ros::NodeHandle& nh) :
+  nh_(nh) {
+  this->loadParams();
+  this->init();
+  this->rosSetup();
+  point_head_client_ =
+    new PointHeadClient("/head_traj_controller/point_head_action", true);
+  speak_client_ = new SpeakClient("/sound_play", true);
+
+  ROS_INFO("Waiting for point head action server...");
+  ROS_WARN("Head will not move if joystick node is up! Please stop it now.");
+  if (!point_head_client_->waitForServer(ros::Duration(2.0))) {
+    ROS_WARN("Couldn't reach the head trajectory controller. Head will not move.");
+  }
+
+  ROS_INFO("Waiting for sound play action server...");
+  if (!speak_client_->waitForServer(ros::Duration(2.0))) {
+    ROS_WARN("Couldn't reach the speaker client. No audio will be played.");
+  }
+  ROS_INFO("PR2 Head initialised.");
+}
+
+Head::~Head() {
+  ros::param::del(nh_.getNamespace());
+  delete point_head_client_;
+  delete speak_client_;
+}
+
+void Head::loadParams() {
+  nh_.getParam("follow_object", follow_object_);
+  nh_.getParam("table_pos/x", table_pos_.x);
+  nh_.getParam("table_pos/y", table_pos_.y);
+  nh_.getParam("table_pos/z", table_pos_.z);
+}
+
+void Head::init() {
+  target_object_ = "";
+  look_at_object_ = "";
+}
+
+void Head::rosSetup() {
+  target_object_sub_ = nh_.subscribe("target_object", 1, &Head::objCB, this);
+  talk_srv_ = nh_.advertiseService("say", &Head::say, this);
+}
+
+void Head::objCB(const std_msgs::String::Ptr msg) {
+  ROS_INFO_STREAM("TargObj: " << target_object_);
+  target_object_ = msg->data;
+  ROS_INFO_STREAM("TargObj: " << target_object_);
+}
+
+void Head::ready() {
+  this->speak("PR2 Ready");
+  this->lookAt("base_link", table_pos_.x, table_pos_.y, table_pos_.z);
+}
+
+void Head::updateLookingPosition() {
+  bool new_obj = target_object_ != look_at_object_;
+  if (target_object_ != "") {
+    if ((new_obj) || (follow_object_ == true) ) {
+      look_at_object_ = target_object_;
+      if (new_obj) {
+        // std::string sentence = "Looking at " + look_at_object_;
```

Comment:
Could you add a ROS param to determine if this commented code gets executed?
Like: if (new_obj && declare_obj) {
UNCOMMENTED LINES}